src/setuptools_scm/git.py

```diff
@@ -82,20 +83,23 @@ def fail_on_shallow(wd):
         )
 
 
-def parse(root, describe_command=DEFAULT_DESCRIBE, pre_parse=warn_on_shallow):
+def parse(root, config=None, describe_command=DEFAULT_DESCRIBE, pre_parse=warn_on_shallow):
     """
     :param pre_parse: experimental pre_parse action, may change at any time
     """
+    if not config:
```

Comment:
i believe have an idea for a followup pr to get the positional arguments and the keyword arguments in a desirable shape for backward compat support, at that point i'll also introduce DeprecationWarnings for the old apis so it can be dropped in 1-2 years